Update the pong tutorial + changelog for SpriteRenderData

Moxinilian opened the pull request:
This PR updates the pong tutorial in the book to have it reflect what is in the repo, after the addition of the `SpriteRenderData` utilities. It also officially adds the utilities to the changelog.

**Could the person who made the example screenshots update them? I have no idea how to make them clean and good looking like that.**

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/amethyst/amethyst/805)
<!-- Reviewable:end -->

Xaeroxe commented on book/src/pong_tutorial/pong_tutorial_02.md:
```diff
@@ -39,20 +39,90 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
+that we the create from an orthographic projection of the size of our
+arena (as we want it to cover it all). Ignore the `GlobalTransform` for now,
+we'll deal with it in more details later on.
+Note that we flipped the direction of the verical dimension by specifying
```
"vertical"

Xaeroxe commented on book/src/pong_tutorial/pong_tutorial_02.md:
````diff
@@ -39,20 +39,90 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
+```
+
+We create an entity that will carry our camera,
+that we the create from an orthographic projection of the size of our
````
Grammar flow here is a bit odd, perhaps omit "that we the create"

Xaeroxe commented on book/src/pong_tutorial/pong_tutorial_02.md:
```diff
@@ -294,85 +314,90 @@ world
+Before we continue, one last note.
+Components do not have to be registered manually to be used, however you need to have
+something that uses them to have them be registered automatically.
+As nothing uses our `Paddle` component yet, we will register it manually before we intialise
```
"initialize"

Moxinilian replied:
There is another instance of "initialise" that I did not write tho lol
I'll fix it too.

Moxinilian replied:
I think that's american vs british here right?

Moxinilian replied:
Ah I'm stupid nvm

Xaeroxe replied:
The real reason this caught my attention was the missing second "i" so remember to get that too.

Xaeroxe commented on examples/pong_tutorial_01/main.rs:
```diff
@@ -27,9 +27,13 @@ fn main() -> amethyst::Result<()> {
         env!("CARGO_MANIFEST_DIR")
     );
 
+    // This line is not mentionned in the pong tutorial as it is specific to the context
```
"mentioned"

Xaeroxe replied:
@Moxinilian latest commit missed this one :P

Moxinilian replied:
damn it

Xaeroxe commented on examples/pong_tutorial_02/main.rs:
```diff
@@ -14,10 +14,14 @@ fn main() -> amethyst::Result<()> {
         env!("CARGO_MANIFEST_DIR")
     );
 
+    // This line is not mentionned in the pong tutorial as it is specific to the context
```
"mentioned"

Xaeroxe commented on examples/pong_tutorial_03/main.rs:
```diff
@@ -21,14 +21,18 @@ fn main() -> amethyst::Result<()> {
         env!("CARGO_MANIFEST_DIR")
     );
 
+    // This line is not mentionned in the pong tutorial as it is specific to the context
```
"mentioned"

Xaeroxe commented:
#803 may have a conflict with this, so be sure to rebase once it's merged in an hour or so.

Xaeroxe commented:
@Moxinilian 

> Could the person who made the example screenshots update them? I have no idea how to make them clean and good looking like that.

Alt + Print Screen will probably help, then on Windows all you need to do is paste into something that can accept an image, such as the "Paint" program.

Moxinilian commented:
Yeah I know, but the way they are cut is very clean. It has shading but no background. Plus it was made on macOS.

Moxinilian commented:
Squashing is so easy with branches

jojolepro commented on book/src/pong_tutorial/pong_tutorial_02.md:
```diff
@@ -39,20 +39,89 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
+## Get around the World
+
+First, in `pong.rs`, let's add a new method to our State implementation: `on_start`.
+This method is called, as you probably guessed, on the start of the State.
```
on the start of the State. -> when the State starts

(the important part is the "when")

jojolepro commented on book/src/pong_tutorial/pong_tutorial_02.md:
```diff
@@ -39,20 +39,89 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
+the upper part of the screen, so here it will feel more natural to work with.
+
+> Orthographic projections are a type of 3D visualization on 2D screens
+> that keeps the size ratio of the 2D images displayed intact, very useful in
```
displayed intact, very useful  -> displayed intact. They are very useful

jojolepro commented on book/src/pong_tutorial/pong_tutorial_02.md:
````diff
@@ -132,28 +183,29 @@ to a parent if one exists, while `GlobalTransform` is, well, global.
 ```
 
 Okay, let's make some entities! We'll define an `initialise_paddles` function
 which will create left and right paddle entities and attach `Transform`
````
attach `Transform` components to them -> attach a `Transform` component to each

Moxinilian commented:
I also moved the spritesheet out of `assets/img` into `assets/texture`, I had misjudged the content of `img`.

jojolepro commented:
Fix merge issues and I'll start the merge for you ;)

Xaeroxe commented:
@Moxinilian hey can you rebase this so we can get it merged?

Moxinilian commented:
Sure, as soon as I get back home (in less than 24 hours). 


12. Jul 2018 16:40 by notifications@github.com <mailto:notifications@github.com>:


>
> @Moxinilian <https://github.com/Moxinilian>>  hey can you rebase this so we can get it merged?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, > view it on GitHub <https://github.com/amethyst/amethyst/pull/805#issuecomment-404535982>> , or > mute the thread <https://github.com/notifications/unsubscribe-auth/AdjoJGBHA80-DFf96OjFRxXONrZnipvIks5uF1_rgaJpZM4U2jeb>> .
>

Moxinilian commented:
Okay I rebased everything and fixed a small mistake in tutorial 2 (regarding the explanation of the orthographic projection).
As soon as CI is done, it's ready for merging!

Xaeroxe commented:
Thanks @Moxinilian !

bors r+

jojolepro commented:
bors retry
